### DISCUSSION TITLE
Option to disable anonymous access to workspaces

### DISCUSSION THREAD

**Discussion by duanemck** [structurizr/onpremises#63] [Q&A]
### Description

I'm looking to get the on premises instance up and running in my company but it will be exposed publicly on the internet. 
I'm weary of people accidentally exposing workspaces by default because they don't realise they need to limit access manually. 

I'd love a config property that prevents access until a user is logged in. Ideally right as the page is initially loaded redirect to login if the user is not authenticated.

### Priority

Medium

### Resolution

I'm willing to add this feature myself and raise a PR

### More information

I'd be willing to dig in and try get this working myself but I'm not very familiar with Spring and Spring Security. 
If some guidance can be provided I'll try pick it up, else if someone else is happy to do it that would be great to.

**Comment by simonbrowndotje** [comment]
I'm building a plugin feature for one of our customers because they want to ensure workspace security is consistently applied regardless of what users do. It's a work in progress, and not documented yet, but you can find an example (with some basic instructions) at https://github.com/simonbrowndotje/structurizr-onpremises-plugin-example

**Comment by duanemck** [comment]
Great, thank you that looks very promising.
